Sources/DeepLearning/Layer.swift

```diff
@@ -36,12 +36,16 @@ public protocol Layer: Differentiable & KeyPathIterable
     func applied(to input: Input) -> Output
 }
 
-public extension Layer {
+public extension Layer
+    where TangentVector : AdditiveArithmetic, CotangentVector : AdditiveArithmetic,
+          Input.TangentVector : AdditiveArithmetic, Input.CotangentVector : AdditiveArithmetic {
     func valueWithPullback(at input: Input)
         -> (output: Output,
             pullback: (Output.CotangentVector)
                 -> (layerGradient: CotangentVector, inputGradient: Input.CotangentVector)) {
-        let (out, pullback) = _valueWithPullback(at: self, input, in: Self.applied(to:))
+        let (out, pullback) = valueWithPullback(at: input) { layer, input in
```

Comment:
Just return it directly.

Reply:
Please note that the returned pullback is a labeled tuple: (layerGradient: CotangentVector, inputGradient: Input.CotangentVector)

Reply:
I see. I'd expect tuple coercion to work implicitly, but apparently my memory is from years ago.